File: docs/agentile/specs/0002-spike-selection-and-highlight-path/fixtures/sample.py
"""Sample fixture for the Omasnap highlight-path spike.

Exercises comments, docstrings, f-strings, numbers, decorators, classes,
keywords, builtins and punctuation.
"""

from dataclasses import dataclass
from typing import Optional

MAX_RETRIES: int = 3
PATTERN = r"^[a-z0-9_-]+$"


@dataclass
class Snippet:
    """Represents a themed snippet ready for rendering."""

    source: str
    language: str = "python"

    def __post_init__(self) -> None:
        self._tokens: list[dict] = []

    @classmethod
    def from_file(cls, path: str) -> "Snippet":
        return cls(source=path)

    def tokenize(self) -> list[dict]:
        label = f"tokenizing {self.language}"
        print(label)
        for i in range(MAX_RETRIES):
            ok = bool(self.language)
            self._tokens.append({"index": i, "ok": ok})
        return self._tokens

    @property
    def token_count(self) -> int:
        return len(self._tokens)


def greet(name: str, count: Optional[int] = None) -> str:
    count = count if count is not None else 1 + 2
    return f"Hello, {name}! You have {count} new messages."


def main() -> None:
    with open("fixtures/sample.py", "r", encoding="utf8") as handle:
        contents = handle.read()
    snippet = Snippet(contents)
    snippet.tokenize()
    print(greet("Omasnap"), snippet.token_count, len(contents) > 0)


if __name__ == "__main__":
    try:
        main()
    except Exception as err:  # noqa: BLE001
        print(f"failed: {err}")

File: docs/agentile/specs/0002-spike-selection-and-highlight-path/fixtures/sample.js
// Sample fixture for the Omasnap highlight-path spike.
// Exercises comments, strings, template literals, regex, numbers, classes,
// decorators (as comments, since plain JS lacks them), keywords and builtins.

import { readFile } from "node:fs/promises";

const MAX_RETRIES = 3;
const PATTERN = /^[a-z0-9_-]+$/i;

/**
 * Represents a themed snippet ready for rendering.
 */
export class Snippet {
  #tokens = [];

  constructor(source, language = "javascript") {
    this.source = source;
    this.language = language;
  }

  static fromFile(path) {
    return new Snippet(path);
  }

  async tokenize() {
    const label = `tokenizing ${this.language}`;
    console.log(label);
    for (let i = 0; i < MAX_RETRIES; i++) {
      if (PATTERN.test(this.language)) {
        this.#tokens.push({ index: i, ok: true });
      } else {
        this.#tokens.push({ index: i, ok: false });
      }
    }
    return this.#tokens;
  }

  get tokenCount() {
    return this.#tokens.length;
  }
}

function greet(name) {
  return `Hello, ${name}! You have ${1 + 2} new messages.`;
}

async function main() {
  const contents = await readFile("./fixtures/sample.js", "utf8");
  const snippet = new Snippet(contents);
  await snippet.tokenize();
  console.log(greet("Omasnap"), snippet.tokenCount, contents.length > 0);
}

main().catch((err) => {
  console.error("failed:", err.message);
  process.exitCode = 1;
});
